nomos-blend/message/src/message.rs

```diff
@@ -41,6 +41,16 @@ pub struct PublicHeader {
     pub signature: Signature,
 }
 
+impl PublicHeader {
```

Comment:
Do you think it makes sense to have this return a `VerifiedPublicHeader` that does not have the `signature` field anymore?